packages/core/src/actions/smartAccount/types.ts

```diff
@@ -14,46 +15,51 @@ import type {
 
 //#region UpgradeAccountParams
 export type UpgradeAccountParams<
-  TAccount extends SmartContractAccount | undefined
+  TAccount extends SmartContractAccount | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 > = {
   upgradeTo: UpgradeToData;
-  overrides?: UserOperationOverrides;
+  overrides?: UserOperationOverrides<TEntryPointVersion>;
   waitForTx?: boolean;
 } & GetAccountParameter<TAccount>;
 //#endregion UpgradeAccountParams
 
 //#region SendUserOperationParameters
 export type SendUserOperationParameters<
-  TAccount extends SmartContractAccount | undefined
+  TAccount extends SmartContractAccount | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 > = {
   uo: UserOperationCallData | BatchUserOperationCallData;
-  overrides?: UserOperationOverrides;
+  overrides?: UserOperationOverrides<TEntryPointVersion>;
 } & GetAccountParameter<TAccount>;
 //#endregion SendUserOperationParameters
 
 //#region SignUserOperationParameters
 export type SignUserOperationParameters<
-  TAccount extends SmartContractAccount | undefined
+  TAccount extends SmartContractAccount | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 > = {
-  uoStruct: UserOperationStruct;
+  uoStruct: UserOperationStruct<TEntryPointVersion>;
 } & GetAccountParameter<TAccount>;
 //#endregion SignUserOperationParameters
 
 //#region SendTransactionsParameters
 export type SendTransactionsParameters<
-  TAccount extends SmartContractAccount | undefined
+  TAccount extends SmartContractAccount | undefined,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 > = {
   requests: RpcTransactionRequest[];
-  overrides?: UserOperationOverrides;
+  overrides?: UserOperationOverrides<TEntryPointVersion>;
```

Comment:
since we already have an account param available to us in these requests, then we can infer the `TEntryPointVersion` and remove the need to define it here. you can use `TAccount["entryPoint"]["version"]` if it's defined or use the `account` param from the `GetAccountParameter<TAccount>` if it's not defined. You can have a helper type to this like `GetEntrypointVersionFromAccount` or something. that keeps the EP responsibility on the account itself rather than having to put it on the client too

Reply:
this applies to this whole file, I don't think we need to make these methods generic on `EPVersion`

Reply:
but what about bundler client? that one also needs entry point version, but not attached to account. @moldy530

Reply:
I don't think it does does it? you can make the bundler methods take a union type instead of having to pass in the generic

Reply:
hmm i see, so tldr, client not attached to smart account can be used for either v6 and v7.

Reply:
if it defaults to the union type already, why not just remove the generic altogether? IIRC the way viem handles the various TX types is a union type without using generics

Reply:
because dealing with union types is not ideal for devs, as you have to do lots of repetitive checking for which entry point version struct/response it is, and you have to cast types so makes the code for devs who use SDK very ugly.

I still think we should use generic for better typescript experience vs using union types.

Reply:
when using the `bundlerClient` a dev doesn't need to do type checking though. the method just directly supports either format. Looking at how `viem` handles this here https://github.com/wevm/viem/blob/main/src/actions/wallet/sendTransaction.ts#L49 and here https://github.com/wevm/viem/blob/main/src/types/transaction.ts#L231 they also just use a union type. Where they leverage the generic is if the actual request types require something to be generic (which in our case neither of the 2 types require that). 

I would rather we stay in line with how the bundler handles it too which is to use Union types